Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -521,620 +538,183 @@ void FilterEvents::groupOutputWorkspace() {
-                   [](const std::unique_ptr<TimeSeriesProperty<TYPE>> &x) { return x.get(); });
-    tsp->splitByTimeVector(split_datetime_vec, m_vecSplitterGroup, split_properties);
-  }
+  m_targetWorkspaceIndexSet = m_timeSplitter.outputWorkspaceIndices();
 
-  // assign to output workspaces
-  for (int tindex = 0; tindex <= max_target_index; ++tindex) {
-    // find output workspace
-    auto wsiter = m_outputWorkspacesMap.find(tindex);
-    if (wsiter == m_outputWorkspacesMap.end()) {
-      // unable to find workspace associated with target index
-      g_log.information() << "Workspace target (" << tindex << ") does not have workspace associated."
-                          << "\n";
-    } else {
-      // add property to the associated workspace
-      DataObjects::EventWorkspace_sptr ws_i = wsiter->second;
-      ws_i->mutableRun().addProperty(std::move(output_vector[tindex]), true);
-    }
-  }
-
-  return;
+  // For completeness, make sure we have a special workspace index for unfiltered events
+  m_targetWorkspaceIndexSet.insert(TimeSplitter::NO_TARGET);
 }
 
-//----------------------------------------------------------------------------------------------
-/** Purpose:
- *    Convert SplitterWorkspace object to SplittingIntervalVec (sorted vector)
- *    and create a map for all workspace group number
- *  Requirements:
- *  Gaurantees:
- *  - Update of m_maxTargetIndex: it can be zero in SplittersWorkspace case
- * @brief FilterEvents::processSplittersWorkspace
+//-------------------------------------------------------------------------
+/** Create a list of EventWorkspace objects to be used as event filtering output.
+ * Sets the TimeROI for each destination workspace
  */
-void FilterEvents::processSplittersWorkspace() {
-  // 1. Init data structure
-  size_t numsplitters = m_splittersWorkspace->getNumberSplitters();
-  m_splitters.reserve(numsplitters);
-
-  // 2. Insert all splitters
-  m_maxTargetIndex = 0;
-  bool inorder = true;
-  for (size_t i = 0; i < numsplitters; i++) {
-    // push back the splitter in SplittersWorkspace to list of splitters
-    m_splitters.emplace_back(m_splittersWorkspace->getSplitter(i));
-    // add the target workspace index to target workspace indexes set
-    m_targetWorkspaceIndexSet.insert(m_splitters.back().index());
-    // register for the maximum target index
-    if (m_splitters.back().index() > m_maxTargetIndex)
-      m_maxTargetIndex = m_splitters.back().index();
-    // check whether the splitters are in time order
-    if (inorder && i > 0 && m_splitters[i] < m_splitters[i - 1])
-      inorder = false;
-  }
-  m_progress = 0.05;
-  progress(m_progress);
+void FilterEvents::createOutputWorkspaces() {
+  const auto startTime = std::chrono::high_resolution_clock::now();
 
-  // 3. Order if not ordered and add workspace for events excluded
-  if (!inorder) {
-    std::sort(m_splitters.begin(), m_splitters.end());
+  size_t min_expected_number_of_indexes =
```

Comment:
This appears to be constant. If that is correct, mark it `const` to make it clearer. Otherwise, the comment should be modified to explain how/when it is changed.